zimui/src/plugins/webp-hero.ts

```diff
@@ -1,11 +1,26 @@
-import { WebpMachine, detectWebpSupport } from 'webp-hero'
+import {
+  WebpMachine,
+  detectWebpSupport,
+  loadBinaryData,
+  convertDataURIToBinary,
+  isBase64Url
+} from 'webp-hero'
+import { Webp } from 'webp-hero/libwebp/dist/webp'
 import 'webp-hero/dist-cjs/polyfills.js'
 
-export const triggerWebpPolyfill = () => {
-  detectWebpSupport().then((support_webp) => {
-    if (!support_webp) {
-      const webpMachine = new WebpMachine()
-      webpMachine.polyfillDocument()
+const webp = new Webp()
+
+export async function polyfillThumbnail(src: string): Promise<string> {
+  const isWebpSupport = await detectWebpSupport()
+  try {
+    if (!isWebpSupport) {
+      const wM = new WebpMachine({ webp })
```

Comment:
`webpMachine` was way clearer than `wM` from my perspective. Any reason to prefer `wM`?

Reply:
Changed in [82feca5](https://github.com/openzim/youtube/pull/272/commits/82feca5fdf816dfdf510da9b3a08eae589d2409a)